### PR TITLE
Fix ReadOnlySequence First

### PR DESCRIPTION
**Pull request by benaadams**
Seen in https://github.com/dotnet/corefxlab/pull/2143

Don't have enough bandwidth on the train to download and install a fresh coreclr to get passed apicheck; but the test definitely fails before the change. Hopefully it passes after :)

/cc @ahsonkhan @davidfowl @pakrym

### CODE REVIEW

**Comment by benaadams** [comment]
Centos.74.Amd64.Open:Release-x64 
System.Net.Http.Functional.Tests 🔥

@dotnet-bot test Linux x64 Release Build

**Comment by benaadams** [comment]
@dotnet-bot  Tizen armel Debug Build

**Comment by davidfowl** [comment]
LGTM

File: src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs [hunk 1]
```diff
@@ -103,6 +103,44 @@ public void CheckEndReachableDoesNotCrossPastEnd()
             Assert.Equal(200, seq.Length);
         }
 
+        [Fact]
+        public void CanGetFirst()
```
Comment (davidfowl) [inline src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs]: We have other First tests right?

Comment (benaadams) [reply src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs]: Yes; though not mismatched multi segment buffer sizes (which this is)

**Comment by benaadams** [comment]
Don't have permission to rerun Tizen; and OSX is just generally broken

**Comment by ahsonkhan** [comment]
This file isn't included in the test csproj (and it needs to be): https://github.com/dotnet/corefx/blob/master/src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.TryGet.cs

And it needs to be updated to remove use of IMemoryList!

File: src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs [hunk 2]
```diff
@@ -103,6 +103,44 @@ public void CheckEndReachableDoesNotCrossPastEnd()
...
+        {
+            var bufferSegment1 = new BufferSegment<byte>(new byte[100]);
+            BufferSegment<byte> bufferSegment2 = bufferSegment1.Append(new byte[100]);
+            BufferSegment<byte> bufferSegment3 = bufferSegment2.Append(new byte[100]);
+            BufferSegment<byte> bufferSegment4 = bufferSegment3.Append(new byte[200]);
+
+            var buffer = new ReadOnlySequence<byte>(bufferSegment1, 0, bufferSegment4, 200);
+
+            Assert.Equal(500, buffer.Length);
+            var length = 500;
```
Comment (ahsonkhan) [inline src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs]: nit: fix use of vars

**Comment by ahsonkhan** [comment]
> Don't have permission to rerun Tizen

What do you mean?

@dotnet-bot test Tizen armel Debug Build

**Comment by benaadams** [comment]
dotnet bot ignores me in corefx for Tizen reruns (works in coreclr though). I think its a limited group because it gets so overloaded

**Comment by benaadams** [comment]
image

**Comment by ahsonkhan** [comment]
BuffersExtensions at 70% - we need to get that coverage up - I will add to https://github.com/dotnet/corefx/issues/26768

Anything from the `ROS<T>` coverage (~93%) that leaps out at you that is important to test?

**Comment by benaadams** [comment]
`BuffersExtensions.Write<T>` is probably the main thing with 0% coverage; others don't look too exciting. 

Some argument throws for .ctor validation (empty array). A couple exceptions I'm not sure can ever be hit, some non-segment paths (pure array/string/OwnedMemory cast paths)

**Comment by benaadams** [comment]
added report to issue

File: src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs [hunk 3]
```diff
@@ -103,6 +103,44 @@ public void CheckEndReachableDoesNotCrossPastEnd()
...
+            int length = 500;
+
+            for (int s = 0; s < 3; s++)
+            {
+                for (int i = 100; i > 0; i--)
+                {
+                    Assert.Equal(i, buffer.First.Length);
+                    buffer = buffer.Slice(1);
+                    length--;
+                    Assert.Equal(length, buffer.Length);
```
Comment (pakrym) [inline src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs]: I've would've also checked contents.

Comment (davidfowl) [reply src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs]: @benaadams can you add that?

Comment (benaadams) [reply src/System.Memory/tests/ReadOnlyBuffer/ReadOnlySequenceTests.Common.cs]: Might not be soon. In airport, then in plane for 10 hrs 😀

**Pull request merged**